### PR TITLE
feat/otel support

### PR DESCRIPTION
**Pull request by rbiseck3**
### Description
Use opentelemetry SDK to expose timing information. BY default, this uses the same timing info that was already beign exposed using the console exporter with a custom formatter: 
```
EmbedStep [cls] finished in 4.645677s
```

But now also supports an optional otel endpoint to be supplied to publish the tracing data to. A follow up to this should be to collect timing data for each run and check for any degradation in performance via CI. 

image

To test with otel, run the docker compose introduced in this PR:
```
docker compose -f ./test_e2e/docker-compose.otel.yml up -d
```

This will expose the collector on `http://localhost:4318`, which can be passed in as the `otel_endpoint` parameter to the `ProcessorConfig`. 

View the trace data in the Jaegar UI at http://localhost:16686/.

### CODE REVIEW

File: test_e2e/expected-structured-output/box/nested-1/ideas-page.html.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
...
+      "text_as_html": "<table><tr><td/><td/><td>January 2023 ( Someone fed my essays into GPT to make something that could answer<br/>questions based on them, then asked it where good ideas come from. The<br/>answer was ok, but not what I would have said. This is what I would have said.) The way to get new ideas is to notice anomalies: what seems strange,<br/>or missing, or broken? You can see anomalies in everyday life (much<br/>of standup comedy is based on this), but the best place to look for<br/>them is at the frontiers of knowledge. Knowledge grows fractally.<br/>From a distance its edges look smooth, but when you learn enough<br/>to get close to one, you&#x27;ll notice it&#x27;s full of gaps. These gaps<br/>will seem obvious; it will seem inexplicable that no one has tried<br/>x or wondered about y. In the best case, exploring such gaps yields<br/>whole new fractal buds.</td></tr></table>",
```
[the remark this answers is not in the document]
Comment (potter-potter) [reply test_e2e/expected-structured-output/box/nested-1/ideas-page.html.json:7]: I guess its a self closing table cell.

[the remark this answers is not in the document]
Comment (potter-potter) [reply test_e2e/expected-structured-output/box/nested-1/ideas-page.html.json:7]: But rumor is that its not legit html.

[the remark this answers is not in the document]
Comment (rbiseck3) [reply test_e2e/expected-structured-output/box/nested-1/ideas-page.html.json:7]: Asked @MthwRobinson about this, got the OK to update the fixtures.

**Review by potter-potter — APPROVED** [review]
Works well.

**Pull request merged**